### PR TITLE
Fixing issues with resource parameters

### PR DESCRIPTION
**Pull request by kavishkafernando**
Fixes https://github.com/wso2/product-apim/issues/6772
Fixes https://github.com/wso2/product-apim/issues/6737
Fixes https://github.com/wso2/product-apim/issues/6783

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/AddParameter.jsx [hunk 1]
```diff
@@ -80,6 +84,18 @@ function AddParameter(props) {
     React.useEffect(() => {
         setLabelWidth(inputLabel.current.offsetWidth);
     }, []);
+
+    React.useEffect(() => {
+        const isParameterExistValue = operation.parameters && operation.parameters.map(operations =>
+            (!!((operations.in === newParameter.in && operations.name === newParameter.name))));
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/AddParameter.jsx]: @kavishkafernando , Why do we need double not (**!**) here?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/AddParameter.jsx [hunk 2]
```diff
@@ -80,6 +84,18 @@ function AddParameter(props) {
     React.useEffect(() => {
         setLabelWidth(inputLabel.current.offsetWidth);
     }, []);
+
+    React.useEffect(() => {
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/AddParameter.jsx]: @kavishkafernando , I think we don't need an `effect` here ? we could derive the `isParameterExist` from the `operation` props

Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/AddParameter.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/AddParameter.jsx (hunk 1 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/AddParameter.jsx]: fixed

**Pull request merged**